### PR TITLE
fix some eslint error

### PR DESCRIPTION
**Pull request by minggo**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/assembler/label/font-utils.ts [hunk 1]
```diff
@@ -400,7 +400,7 @@ export class LetterAtlas {
     }
 
     public getLetterDefinitionForChar (char: string, labelInfo: ILabelInfo): any {
-        const hash = char.charCodeAt(0) + labelInfo.hash;
+        const hash = char.charCodeAt(0).toString() + labelInfo.hash;
```
Comment (minggo) [inline cocos/2d/assembler/label/font-utils.ts:403]: As  key of `letterDefinitions ` is string, so use string here.

File: cocos/core/data/utils/preprocess-class.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ function parseType (val, type, className, propName): void {
         if (type.length > 0) {
             val.type = type = type[0];
         } else {
-            return errorID(5508, className, propName);
+            errorID(5508, className, propName);
```
Comment (SantyWang) [inline cocos/core/data/utils/preprocess-class.ts:95]: Should keep `return` logic;

File: cocos/audio/audio-manager.ts [hunk 3]
```diff
@@ -97,6 +97,7 @@ export class AudioManager {
             });
         }
         if (audioInfoToDiscard) {
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
             audioInfoToDiscard.audio.stop();
```
Comment (PPpro) [inline cocos/audio/audio-manager.ts]:
```suggestion
            audioInfoToDiscard.audio.stop().catch((err) => {});
```

it‘s better to catch the error of promise

File: cocos/core/data/utils/preprocess-class.ts (hunk 2 above)
Comment (minggo) [reply cocos/core/data/utils/preprocess-class.ts:95]: The return type is void.

Comment (SantyWang) [reply cocos/core/data/utils/preprocess-class.ts:95]: I know, but we should not execute the code below.

**Review by SantyWang — APPROVED** [review]

File: cocos/audio/audio-manager.ts (hunk 3 above)
Comment (minggo) [reply cocos/audio/audio-manager.ts]: fixed

**Review by PPpro — APPROVED** [review]

**Pull request merged**